lib/presentation/search/search_page.dart

```diff
@@ -10,105 +15,117 @@ import 'widgets/current_location_widget.dart';
 import 'widgets/past_search_block.dart';
 import 'widgets/search_text_field.dart';
 
-class SearchPage extends StatefulWidget {
+class SearchPage extends HookWidget {
   const SearchPage({
     Key? key,
-    required this.chosenCity,
     required this.pastSearchCities,
-    required this.onCityChosen,
   }) : super(key: key);
-
-  final ValueNotifier<String> chosenCity;
-  final List<String> pastSearchCities;
-  final VoidCallback onCityChosen;
+  final ValueNotifier<List<String>> pastSearchCities;
 
   @override
-  State<SearchPage> createState() => _SearchPageState();
-}
+  Widget build(BuildContext context) {
+    final context = useContext();
+    final controller = useTextEditingController();
+    final focusNode = useFocusNode();
+    useListenable(focusNode);
+    useListenable(pastSearchCities);
+    useListenable(controller);
 
-class _SearchPageState extends State<SearchPage> {
-  final _controller = TextEditingController();
-  final _focusNode = FocusNode();
+    // ignore: prefer_function_declarations_over_variables
+    final onCityItemTap = (String city) {
+      pastSearchCities.value =
+          List.unmodifiable([...pastSearchCities.value, city]);
+      context.dispatch(GetWeatherByCityNameAction(cityName: city));
+      focusNode.unfocus();
+      appRouter.goBack(context);
+    };
+
+    // ignore: prefer_function_declarations_over_variables
+    final onEditingComplete = ({
+      required String cityName,
+    }) {
+      if (cityName.isNotEmpty) {
+        pastSearchCities.value =
+            List.unmodifiable([...pastSearchCities.value, cityName]);
+        context.dispatch(GetWeatherByCityNameAction(cityName: cityName));
+        appRouter.goBack(context);
+      }
+    };
 
-  @override
-  Widget build(BuildContext context) {
     return Scaffold(
-      body: Padding(
-        padding: const EdgeInsets.symmetric(horizontal: 24),
-        child: Column(
-          crossAxisAlignment: CrossAxisAlignment.start,
-          children: [
-            const SizedBox(height: 65),
-            IconButton(
-              onPressed: () => AppRouter().goBack(context),
-              icon: Image.asset(
-                Images.icClose,
-                width: 32,
-                height: 32,
+      body: AppStateConnector<AppState>(
+        converter: (state) => state,
```

Comment:
```suggestion
        converter: (state) => state.weather.currentCity,
```

судя по коду ниже, тут должно быть так)
если подпишешься на весь стейт, то любое его изменение приведет к перерисовке всего экрана, а тут нужно отслеживать только текущий город